src/api/middleware/utils.js

```diff
@@ -0,0 +1,56 @@
+const HTTPStatus = require("http-status-codes");
+const { APIError, ErrorTypes } = require("./errorHandler");
+const ValidationReasons = require("./validators/validationReasons");
+const lodash = require("lodash");
+
+const DEFAULT_ERROR_CODE = ErrorTypes.VALIDATION_ERROR;
+const DEFAULT_ERROR_MSG = ValidationReasons.UNKNOWN;
+const DEFAULT_STATUS_CODE = HTTPStatus.BAD_REQUEST;
+
+/**
+ * Comibnes array of middleware using OR logic. Only fails if ALL functions fail (either by throwing or calling next(error))
+ *
+ * Each middleware will receive a different req object, no not rely on it to be shared among them
+ *
+ * @param {Function[]} Array of express middleware to be run
+ * @param {object} Options:
+ *  - error_code: error code in case of error (default: ErrorTypes.VALIDATION_ERROR)
+ *  - msg: the message in case of error (default: ValidationReasons.UNKNOWN)
+ *  - status_code: The status used in the HTTP Response in case of error (default: BAD_REQUEST (400))
+ */
+const or = (
+    [...middlewares],
+    {
+        error_code = DEFAULT_ERROR_CODE,
+        msg = DEFAULT_ERROR_MSG,
+        status_code = DEFAULT_STATUS_CODE
+    } = {}
+) => async (initialReq, res, next) => {
+    let success = false;
+    const errors = [];
+    for (const middleware of middlewares) {
+        const req = lodash.cloneDeep(initialReq);
+        if (success) return next();
+        try {
+            await middleware(req, res, (error) => {
+                if (error) errors.push(error);
```

Comment:
Errors might not be `APIError` so we cannot call .toObject() on every error. On the current version, that would be equivalent, but I'm thinking we might want to only show errors as is if they are APIError instances, and unknown errors otherwise, to protect leaks.

In this case, we need to first check if the error is an APIError, and in that case push `err.toObject()`, otherwise push a generic unknown error .toObject so that db errors or similar end up leaking.

Reply:
I don't understand then why calling map.((e)=>e.toObject()) worked

Reply:
That's the thing, it won't ahaha

I am making the change now, and will call the toObject inside the for loop, knowing that we'll only deal with APIErrors there. If they are not APIErrors, they will be converted, and toObject can be safely called

Reply:
Oh ok, I didn't realize it in the previous reply